### DISCUSSION TITLE
New/More Content

### DISCUSSION THREAD

**Discussion by neises** [nasosi/C4EngineExamples#4] [General]
Hey nasosi, 

are you planing to add additiona content to the examples?

Cheers

**Comment by nasosi** [comment]
Hello @neises,

I plan to, but this is not my high priority right now. I will be doing it as I find things that are interesting.  If you have any requests post them in the discussions, since they may give me ideas.

Generally I am open to accepting contributions though.

 I am working on a big uplift to the examples build system that I will hopefully finish in the next few  weeks.

**Reply by neises** [reply]
Thanks! A few things that currently pop into my head would be

- Import models (with automatically created material and attached to it - the current documentation is lacky and for some reason I´m not able to import my models correct WITH attached materials)
- How to use Threads in the C4 Engine 
- How to use Shaderparams or pass values to the Shader
- I have to think of other examples

Another question I have is, since you are active on the C4 forum as well and I saw you posted an integration of VR into the 4C Engine - are you planing to make the plugin/integration public. I´m nosy about that one.

thanks in advanced

**Reply by nasosi** [reply]
Hello,
These are some good suggestions. I will probably create some examples after I finish with the changes I am making.

I am not planning to share the VR implementation yet for several reasons. I will reassess when it will reach a maturity level I am comfortable with. Sorry!

**Reply by neises** [reply]
Sounds good. Thanks!

**Reply by neises** [reply]
Oh, I have another one - thats  more a personal request. Maybe you know how to deal with MarkingEffects on Models :-)?

Still trying to figure out how to make this work.

**Comment by neises** [comment]
Hey, @nasosi are you still planing to update and release new examples?

**Reply by nasosi** [reply]
Hello,

Yes I am, but I am working on a distribution system that the examples will also be released through, and needs to be finished first. The purpose is to get rid of cmake and have games/libraries/plugins/and assets packages that more naturally integrate with the engine. I am hoping for an early release in the next few months.

**Reply by neises** [reply]
that sounds awesome and i´m curious to see what the content will be. thanks for this!
